### ISSUE TITLE
Val_acc and Train_acc is very different for same dataset

### ISSUE THREAD

**Issue by mburaksayici** [keras-team/keras#9588]
Hi everyone.  I'm finetuning several networks and I felt some strangeness in val_acc and train_acc. To test it, I used exactly the same dataset for train_acc and val_acc with no process on them. Same numpy file. However, I get train_acc 1.0 and val_acc 0.2593 for exactly the same dataset. I couldn't get it. Overfitting is not the case for me at that situation. I normally use augmentation and class_weights. Here's the code. Also, when I make my model(that I train on my dataset with train_acc 1, val_acc 0.2593 for exactly the same dataset) predict, it always predict the same class. So how does it show that the training_acc is 1 for the dataset that I used also for validation data and get 0.25 val_acc?
I felt something wrong in Keras, so I needed to ask it here. Here's the code and results.
# giris is my images, cikis is the output.

2/1 [===================================] - 7s 3s/step - loss: 12.6554 - categorical_accuracy: 0.8935 - val_loss: 21.2233 - val_categorical_accuracy: 0.2593
--

`print("s")
import keras
from keras.layers import Input
from keras.models import Model, Sequential
from keras.layers.core import Reshape, Dense, Dropout, Flatten
from keras.layers.advanced_activations import LeakyReLU, regularizers
from keras.layers.convolutional import Conv2D, UpSampling2D,MaxPooling2D, Convolution2D
from keras.datasets import mnist
from keras.optimizers import Adam, SGD,RMSprop, Adadelta
from keras import backend as K
from keras.losses import mean_absolute_error
from keras import initializers
from tensorflow import losses as ls
from keras.preprocessing import image
from keras.models import Model
from keras.layers import Dense, GlobalAveragePooling2D
from keras import backend as K
from keras import callbacks
from keras.metrics import top_k_categorical_accuracy,categorical_accuracy
import numpy as np

from keras.applications.vgg16 import VGG16
from keras.preprocessing import image
from keras.models import Model
from keras.layers import Dense, GlobalAveragePooling2D
from keras import backend as K
import imgaug
import augmentfunctions as af
from keras import regularizers
from keras.applications import inception_v3, resnet50
from keras.callbacks import ReduceLROnPlateau

giris = af.giris.reshape(-1,224,224,3)
cikis = af.cikis.reshape(-1,5)

# create the base pre-trained model
base_model = resnet50.ResNet50(weights="imagenet", include_top=False)

# add a global spatial average pooling layer
x = base_model.output
x = GlobalAveragePooling2D()(x)
# let's add a fully-connected layer
x = Dropout(0.5)(x)
x = Dense(1024, activation='relu',kernel_regularizer=regularizers.l2(0.01))(x)

predictions = Dense(5,activation="softmax")(x)

# this is the model we will train
model = Model(inputs=base_model.input, outputs=predictions)

for layer in base_model.layers:
    layer.trainable = False

# compile the model (should be done *after* setting layers to non-trainable)
model.compile(optimizer=keras.optimizers.Adam(lr=0.0001), loss='categorical_crossentropy',metrics=["categorical_accuracy"])

# train the model on the new data for a few epochs
model.fit(giris/255,cikis,epochs=10,validation_data=(giris/255,cikis))

- [ +] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [ +] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ Not needed] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by KiraMichiru** [comment]
```
model.fit_generator(af.generator(giris, cikis, 128), steps_per_epoch=af.giris.shape[0] / 128, epochs=12,validation_data=(giris/255,cikis)
)
```

Is your generator also dividing the giris by 255? Otherwise you feed in different data for training and validation. Also, from the variable naming it seems your generator is augmenting images, so technically you are not feeding the exactly same images.

**Comment by mburaksayici** [comment]
Sorry, I'll edit this, this is the different python file. I'm doing same normally. And yes, in this condition, generator divides it with 255. But for validation part, it doesn't, I do manually.

**Comment by IRiViI** [comment]
Have you found the reason for this? I tried transfer learning using VGG16 and ResNet50 and I had the same problem as you when I used ResNet50 (VGG16 went well).
